server/hw_controller/feeder.py

```diff
@@ -483,10 +508,29 @@ def _parse_device_line(self, line):
             # Marlin "unknow command"
             elif "echo:Unknown command:" in line:
                 self.logger.error("Error: command not found. Can also be a communication error")
+            # M114 response contains the "Count" word
+            # the response looks like: X:115.22 Y:116.38 Z:0.00 E:0.00 Count A:9218 B:9310 Z:0
+            elif "Count" in line:
+                l = line.split(" ")
+                x = float(l[0][2:])     # remove "X:" from the string
+                y = float(l[1][2:])     # remove "Y:" from the string
+                # if the last commanded position coincides with the current position it means the buffer on the device is empty (could happen that the position is the same between different points but the M114 command should not be that frequent to run into this problem.) TODO check if it is good enough or if should implement additional checks like a timeout
+                # need to compare the floor value with 2 decimals (marlin is rounding 0.15 to 0.1 but 0.151 to 0.2)
+                if (round(float(self.last_commanded_position.x),2) == x) and (round(float(self.last_commanded_position.y),2) == y):
```

Comment:
Still not convinced about this solution. If the check is done during a drawing with a buffer stuck and the position is in the wrong round condition it will not pass the check and clear an ack of the buffer. Need to think about the round problem. May use a difference in position below +-0.01 instead of the equlity?
(sometimes marlin is sending a different 2 decimals rounded value with respect to the one obtained with round(value,2) in python, I was not able to identify why)